packages/core/src/lib/lit-core.ts

```diff
@@ -136,17 +152,87 @@ export class LitCore {
   };
 
   /**
+   * Asynchronously updates the configuration settings for the LitNodeClient.
+   * This function fetches the minimum node count and bootstrap URLs for the
+   * specified Lit network. It validates these values and updates the client's
+   * configuration accordingly. If the network is set to 'InternalDev', it
+   * dynamically updates the bootstrap URLs in the configuration.
+   *
+   * @throws Will throw an error if the minimum node count is invalid or if
+   *         the bootstrap URLs array is empty.
+   * @returns {Promise<void>} A promise that resolves when the configuration is updated.
+   */
+  setNewConfig = async (): Promise<void> => {
+    const minNodeCount = await LitContracts.getMinNodeCount(this.config.litNetwork as LitNetwork);
+    const bootstrapUrls = await LitContracts.getValidators(this.config.litNetwork as LitNetwork);
+
+    if (minNodeCount <= 0) {
+      throwError({
+        message: `minNodeCount is ${minNodeCount}, which is invalid. Please check your network connection and try again.`,
+        errorKind: LIT_ERROR.INVALID_ARGUMENT_EXCEPTION.kind,
+        errorCode: LIT_ERROR.INVALID_ARGUMENT_EXCEPTION.name,
+      });
+    }
+
+    if (bootstrapUrls.length <= 0) {
+      throwError({
+        message: `bootstrapUrls is empty, which is invalid. Please check your network connection and try again.`,
+        errorKind: LIT_ERROR.INVALID_ARGUMENT_EXCEPTION.kind,
+        errorCode: LIT_ERROR.INVALID_ARGUMENT_EXCEPTION.name,
+      });
+    }
+
+    // -- Update config
+    // TODO TEMPORARY: only dynamically update when it's set to internalDev
+    if (this.config.litNetwork === LitNetwork.InternalDev) {
+      this.config.bootstrapUrls = bootstrapUrls;
+    }
+
+    this.config.minNodeCount = minNodeCount;
+  }
+
+  /**
+   * Sets up a listener to detect state changes (new epochs) in the staking contract.
+   * When a new epoch is detected, it triggers the `setNewConfig` function to update
+   * the client's configuration based on the new state of the network. This ensures
+   * that the client's configuration is always in sync with the current state of the
+   * staking contract.
    *
-   * Connect to the LIT nodes
-   *
-   * @returns { Promise } A promise that resolves when the nodes are connected.
-   *
+   * @returns {Promise<void>} A promise that resolves when the listener is successfully set up.
+   */
+  listenForNewEpoch = async (): Promise<void> => {
+    const stakingContract = await LitContracts.getStakingContract(this.config.litNetwork as LitNetwork);
+
+    stakingContract.on("StateChanged", async () => {
```

Comment:
I dont know if we want to reload config on every state change
```
    enum States {
        Active,
        NextValidatorSetLocked,
        ReadyForNextEpoch,
        Unlocked,
        Paused,
        Restore
    }
```
I think `nextValidatorSetLocked` will be a good state to check for as then we know the next epoch set is locked in. but this might mean that we should check if the `ValidatorsStructsInCurrentEpoch` are set before attempting to reload the config as we might still be performing dkg and not ready to switch them out.